TutorLizard.BusinessLogic/Data/DataAccess.cs

```diff
@@ -16,6 +16,8 @@ public class DataAccess : IUserIdentityDataAccess, IStudentDataAccess, ITutorDat
     private string _adRequestFilePath = @"Data/ad_requests.json";
     private string _scheduleItemRequestFilePath = @"Data/schedule_requests.json";
 
+    private User _currentUser = new();
```

Comment:
I think that this is more of `UserIdentityService`'s responsibility to keep track of current user. I'd move this functionality there, because it's not strictly connected with reading and writing data.